pkg/worker/worker_test.go

```diff
@@ -0,0 +1,174 @@
+package worker
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	googleUUID "github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+type MemoryStorage struct {
```

Comment:
This is useful outside of tests too.
Please move it to a non-test file and split it into a memory store and mock store (which can inject errors)